### PR TITLE
SiteManager Dashboard: Added Filtration & Pagination on Participant All table

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
This PR addresses the following features:
-> Integrated Filtration & Pagination in to Participants All table
-> Utilizes state manager to retain previously selected filters & provide a robust user experience 
-> User can search participants on verification date, recruitment type & by site
-> Lets users paginate back & forward to view next/previous set of users

Checklist:
- [X] Code cleanup
- [X] Check for ES Lint warnings
- [X] Verify test cases
- [X] Check for GIT conflicts
- [ ] Verified unit tests pass with current changes
- [ ] Any dependencies or modules added
- [X] Attach PoC
- [X] Add notes

Test Cases Verified: 
[FiltrationPaginationTestCases.docx](https://github.com/episphere/dashboard/files/10561120/FiltrationPaginationTestCases.docx)

POC: Available in test case file

### CODE REVIEW

**Comment by we-ai** [comment]
I see that filtration by site, recruitment type and verification data only occurs when "All Participants" is selected in nav bar. If other values (eg. "Verified Participants") are selected in nav bar, there are no filters to select from. I'm not sure whether these're the designed behaviors.

File: siteManagerDashboard/participantCommons.js [hunk 1]
```diff
@@ -677,35 +1026,133 @@ export const renderLookupSiteDropdown = () => {
...
 
 }
 
+const getMoreParticipants = async (query, nextPageCounter) => {
+    let filterHolder = appState.getState().filterHolder
+    if (filterHolder.source === `participantAll`) appState.setState({filterHolder:{...filterHolder, 'nextPageCounter': nextPageCounter}})
+    const siteKey = await getAccessToken();
+    let template = `/dashboard?api=getParticipants`;
+    const limit = 5;
+    console.log('filterHolder.source', filterHolder.source)
```
Comment (we-ai) [inline siteManagerDashboard/participantCommons.js:1043]: Temporary console.log() is not needed in production.

**Comment by abhinavjonnada82** [comment]
> I see that filtration by site, recruitment type and verification data only occurs when "All Participants" is selected in nav bar. If other values (eg. "Verified Participants") are selected in nav bar, there are no filters to select from. I'm not sure whether these're the designed behaviors.

For time being, Filtration & Pagination is only limited to filtration & pagination

**Review by we-ai — APPROVED** [review]
Looks good.

**Pull request merged**